src/gui/FileBrowser.cpp

```diff
@@ -827,7 +827,6 @@ bool Directory::addItems(const QString & path )
 			{
 				addChild( new Directory( cur_file, path,
 								m_filter ) );
-				m_dirCount++;
```

Comment:
Even though the `--- Factory files ---` delimiter comes out in the right spot this way I don't find it a satisfying solution. The orphans seem to be counted twice and if this is the case we should make the logic right instead. I don't know under what circumstances it could pose a problem but the `insertChild()` function may not actually run if it finds the child in question already exists and in this case m_dirCount will still be increased. The same is probably true for `addChild()` as it's realised using `insertChild()`. I guess if this would ever happen we would see actual duplicates in the tree and this doesn't seem to be the case.

Reply:
@zonkmachine What do you mean by "the orphans seem to be counted twice"?

Reply:
As I remember it I had some debug statements in there while testing. Unfortunately I may have to repeat this test in order to understand my own comment.

Reply:
When will you have time for it?

Reply:
Maybe a week.

Reply:
Sorry! I tried to reproduce the debug session but failed. I retraced my steps with printf() debug statements but failed to arrive at the same conclusion. I'm currently at 'I poked that code and it works.' You deserve a better answer... :disappointed:

Reply:
Np, but can you explain if you find your solution still unsatisfying, and if yes, why you find so?

Reply:
I'm just unsatisfied with not knowing precisely what I've done. That's all. I knew the math was wrong and roughly where the error was occurring. I removed that line to poke the bug and it turned out to fix it.